### PR TITLE
Update docker install guide

### PR DESCRIPTION
**Pull request by mooreds**
Was reviewing the docker install guide in the context of some other doco, wanted to make a few changes.

This change:

* reworks some sections
* moves some sections around
* adds examples of adding in additional containers for docker compose
* future proofs some elasticsearch references.

### CODE REVIEW

**Comment by robotdan** [comment]
Un-related to the copy changes, I'll just voice my opinion on how we are using ASCII doc. 

It is super nice to consolidate and re-use components, but if we ever want to use a new framework, it is going to be super hard to unwind a lot of this because it is getting hard to do a find for a piece of text because we use so many replacements and variables. This may get really hard to maintain, and requires quite a bit of ASCIIDOC knowledge to even reason through how all of it goes together.

**Comment by mooreds** [comment]
Definitely appreciate the perspective. The alternative is outdated, inconsistent doco, or requiring users to bounce between links.

For example, the elasticsearch supported versions are currently different on the docker page and the system requirements page. Why? Because when someone updated the system requirements Elasticsearch versions, they forgot to update the docker page.

I don't know what the right answer is. I agree that a move from asciidoc for our docs would be a big effort.

File: site/docs/v1/tech/installation-guide/_docker-compose-files.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
+ifdef::with_elasticsearch[]
+To use FusionAuth with Elasticsearch, download the `docker-compose.yml`, `docker-compose.override.yml` and the `.env` files and then start up the configured containers.
+endif::[]
+ifndef::with_elasticsearch[]
+To use FusionAuth with Elasticsearch, download the `docker-compose.yml` and the `.env` files and then start up the configured containers.
```
Comment (spwitt) [inline site/docs/v1/tech/installation-guide/_docker-compose-files.adoc:6]:
```suggestion
To use FusionAuth without Elasticsearch, download the `docker-compose.yml` and the `.env` files and then start up the configured containers.
```

File: site/docs/v1/tech/installation-guide/docker.adoc [hunk 2]
```diff
@@ -10,102 +10,114 @@ navcategory: installation
...
+* database 
+* search 
+* FusionAuth
+
+Let's look at each of these.
 
 === Database Service
 
-At a minimum, you will need to either set the `POSTGRES_PASSWORD` environment variable in the `db` service section, or more ideally set the value in the host environment and leave it out of the `docker-compose.yml` file. Ensure the other properties fit your requirements. Refer to the link:/docs/v1/tech/installation-guide/system-requirements[System Requirements] for database version support.
+The database service provides a postgreSQL database for use by FusionAuth.
```
Comment (spwitt) [inline site/docs/v1/tech/installation-guide/docker.adoc:79]:
```suggestion
The database service provides a PostgreSQL database for use by FusionAuth.
```
per https://www.postgresql.org/

File: site/docs/v1/tech/installation-guide/docker.adoc [hunk 3]
```diff
@@ -10,102 +10,114 @@ navcategory: installation
...
+
+You will need to either set the `POSTGRES_PASSWORD` environment variable in the `db` service section, or more ideally, set the value in the host environment and leave it out of the `docker-compose.yml` file.
```
Comment (spwitt) [inline site/docs/v1/tech/installation-guide/docker.adoc:81]:
```suggestion
You will need to either set the `POSTGRES_PASSWORD` environment variable in the `db` service section, or, more ideally, set the value in the host environment and leave it out of the `docker-compose.yml` file.
```

File: site/docs/v1/tech/installation-guide/docker.adoc [hunk 4]
```diff
@@ -10,102 +10,114 @@ navcategory: installation
...
-For example if startup is failing and you see the following in the logs, you will need to increase `vm.max_map_count` on your host VM.
+Review the link:/docs/v1/tech/reference/configuration#options[Configuration] documentation to customize your deployment. The best way to configure FusionAuth when using Docker is to use environment variables as documented in that link.
+
+In addition, the port `9011` is mapped between the container and your host machine. This may be modified if you want to run more than one FusionAuth instance or for any other reason.
+
+=== Other Services
+
+You may add other services to the docker compose files if needed.
+
+For example, in development environments it can be helpful to run mailcatcher for SMTP.
```
Comment (spwitt) [inline site/docs/v1/tech/installation-guide/docker.adoc:105]: MailCatcher has capitalization in its name. Should we provide a link to the site?

File: site/docs/v1/tech/installation-guide/docker.adoc [hunk 5]
```diff
@@ -162,23 +174,28 @@ If the installation is in `production` mode, apply the migrations out of band.
...
+
+[source,yaml,title=The FusionAuth Docker file]
+----
+include::https://raw.githubusercontent.com/FusionAuth/fusionauth-containers/master/docker/fusionauth/fusionauth-app/Dockerfile[]
+----
 
-```
-docker pull fusionauth/fusionauth-app
-```
+FusionAuth supports multiple architectures which may not be the case for your needs. There is a build step and then an image creation step.
```
Comment (spwitt) [inline site/docs/v1/tech/installation-guide/docker.adoc:196]: I don't quite follow. I think this is saying that FusionAuth supports multiple architectures, but the supported architectures may not meet the customer's needs? Is the reference to a "build step" and "image creation step" meant to provide a hint as to where a customer could customize to fit their architecture?

I think that "may not be the case" doesn't quite fit here. It's always the case that FusionAuth supports multiple architectures. I'm not sure what a replacement should be without understanding the context here.

File: site/docs/v1/tech/installation-guide/docker.adoc [hunk 6]
```diff
@@ -10,102 +10,114 @@ navcategory: installation
 
 FusionAuth Docker containers can be used with Docker Compose, Kubernetes, Helm or OpenShift. 
 
-The following example is using Docker Compose but you may find links and examples for kubernetes, Helm and OpenShift in the https://github.com/FusionAuth/fusionauth-containers[FusionAuth Containers] GitHub repo.
+The following example is using Docker Compose. Here are link:/docs/v1/tech/installation-guide/kubernetes/[kubernetes and Helm installation instructions. Find links for OpenShift and other community supported Docker environments in the https://github.com/FusionAuth/fusionauth-contrib[FusionAuth Contrib] GitHub repo.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/installation-guide/docker.adoc:13]: Thanks! We had a longstanding goal of moving everything out of fusionauth-containers that is a community contribution. I just did that: https://github.com/FusionAuth/fusionauth-containers/pull/81 so fusionauth-contrib is the right place.

File: site/docs/v1/tech/installation-guide/docker.adoc (hunk 4 above)
Comment (mooreds) [reply site/docs/v1/tech/installation-guide/docker.adoc:105]: Good catch

File: site/docs/v1/tech/installation-guide/docker.adoc [hunk 7]
```diff
@@ -10,102 +10,114 @@ navcategory: installation
 
 FusionAuth Docker containers can be used with Docker Compose, Kubernetes, Helm or OpenShift.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/installation-guide/docker.adoc:11]: Awesome, thanks! I corrected the capitalization and updated our standards doc.

**Review by spwitt — APPROVED** [review]

**Pull request merged**